File: DemoServer/AdditionalLanguages/go/AutoIndexes/autoMapIndex2/AutoMapIndex2.go
package main

//region Usings
import "github.com/ravendb/ravendb-go-client"
//endregion

var globalDocumentStore *ravendb.DocumentStore

func main() {
    createDocumentStore()
    createDatabase()
    autoMapIndex("Steven")
    globalDocumentStore.Close()
}

func createDocumentStore() (*ravendb.DocumentStore, error) {
    if globalDocumentStore != nil {
        return globalDocumentStore, nil
    }
    urls := []string{"http://localhost:8080"}
    store := ravendb.NewDocumentStore(urls, "testGO")
    err := store.Initialize()
    if err != nil {
        return nil, err
    }
    globalDocumentStore = store
    return globalDocumentStore, nil
}

func createDatabase() {
    databaseRecord := ravendb.NewDatabaseRecord()
    databaseRecord.DatabaseName = "testGO"
    createDatabaseOperation := ravendb.NewCreateDatabaseOperation(databaseRecord, 1)
    var err = globalDocumentStore.Maintenance().Server().Send(createDatabaseOperation)
    if err != nil {
        fmt.Printf("d.store.Maintenance().Server().Send(createDatabaseOperation) failed with %s\n", err)
    }
}

//region Demo
func autoMapIndex2(country string) error {
  
    session, err := globalDocumentStore.OpenSession("")
    if err != nil {
        return err
    }
    defer session.Close()

    //region Step_1
    queriedType := reflect.TypeOf(&Employee{})
    query := session.QueryCollectionForType(queriedType)
    query = query.WhereStartsWith("Title", "Sales")
    query = query.Where("Address.Country", "==", country)
    //endregion

    //region Step_2
    var employeeResult *Employee
    err = query.First(&employeeResult)
    if err != nil {
        return err
    }
    //endregion
    
    return nil
}
//endregion

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/autoIndexes/autoMapIndex2/AutoMapIndex2.java
package net.ravendb.demo.autoIndexes.autoMapIndex2;

import net.ravendb.client.documents.session.IDocumentQuery;
import net.ravendb.client.documents.session.IDocumentSession;
import net.ravendb.demo.common.DocumentStoreHolder;
import net.ravendb.demo.common.models.Company;
import net.ravendb.demo.common.models.Employee;

public class AutoMapIndex2 {

    public void run(RunParams runParams) {
        String country = runParams.getCountry();

        //region Demo
        Employee employeeResult;

        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
            //region Step_1
            IDocumentQuery<Employee> findEmployeeQuery = session.query(Employee.class)
                .whereEquals("Address.Country", country)
                .whereStartsWith("Title", "Sales");
            //endregion

            //region Step_2
            employeeResult = findEmployeeQuery.firstOrDefault();
            //endregion
        }
        //endregion
    }

    public static class RunParams {
        private String country;

        public String getCountry() {
            return country;
        }

        public void setCountry(String country) {
            this.country = country;
        }
    }
}
